modules/riseBidAdapter.js

```diff
@@ -239,122 +242,155 @@ function getDeviceType(ua) {
   return '1';
 }
 
+function generateBidsParams(validBidRequests, bidderRequest) {
+  const bidsArray = [];
+
+  if (validBidRequests.length) {
+    validBidRequests.forEach(bid => {
+      bidsArray.push(generateBidParameters(bid, bidderRequest));
+    });
+  }
+
+  return bidsArray;
+}
+
 /**
- * Generate query parameters for the request
- * @param bid {bid}
- * @param bidderRequest {bidderRequest}
- * @returns {Object}
+ * Generate bid specific parameters
+ * @param {bid} bid
+ * @param {bidderRequest} bidderRequest
+ * @returns {Object} bid specific params object
  */
-function generateParameters(bid, bidderRequest) {
+function generateBidParameters(bid, bidderRequest) {
   const {params} = bid;
+  const mediaType = isBanner(bid) ? BANNER : VIDEO;
   const timeout = config.getConfig('bidderTimeout');
-  const {syncEnabled, filterSettings} = config.getConfig('userSync') || {};
-  const [width, height] = getSizes(bid);
-  const {bidderCode} = bidderRequest;
-  const domain = window.location.hostname;
-
-  // fix floor price in case of NAN
-  if (isNaN(params.floorPrice)) {
-    params.floorPrice = 0;
-  }
+  const sizesArray = getSizesArray(bid, mediaType);
 
-  const requestParams = {
-    wrapper_type: 'prebidjs',
-    wrapper_vendor: '$$PREBID_GLOBAL$$',
-    wrapper_version: '$prebid.version$',
-    adapter_version: ADAPTER_VERSION,
-    auction_start: timestamp(),
+  const bidObject = {
+    mediaType,
     ad_unit_code: getBidIdParameter('adUnitCode', bid),
     tmax: timeout,
-    width: width,
-    height: height,
-    publisher_id: params.org,
-    floor_price: Math.max(getFloor(bid), params.floorPrice),
-    ua: navigator.userAgent,
+    sizes: sizesArray,
+    floor_price: Math.max(getFloor(bid, mediaType), params.floorPrice),
     bid_id: getBidIdParameter('bidId', bid),
     bidder_request_id: getBidIdParameter('bidderRequestId', bid),
     transaction_id: getBidIdParameter('transactionId', bid),
-    session_id: getBidIdParameter('auctionId', bid),
-    publisher_name: domain,
-    site_domain: domain,
-    dnt: (navigator.doNotTrack == 'yes' || navigator.doNotTrack == '1' || navigator.msDoNotTrack == '1') ? 1 : 0,
-    device_type: getDeviceType(navigator.userAgent)
   };
 
-  const userIdsParam = getBidIdParameter('userId', bid);
-  if (userIdsParam) {
-    requestParams.userIds = JSON.stringify(userIdsParam);
-  }
-
-  const ortb2Metadata = config.getConfig('ortb2') || {};
-  if (ortb2Metadata.site) {
-    requestParams.site_metadata = JSON.stringify(ortb2Metadata.site);
-  }
-  if (ortb2Metadata.user) {
-    requestParams.user_metadata = JSON.stringify(ortb2Metadata.user);
+  // fix floor price in case of NAN
+  if (isNaN(bidObject.floor_price)) {
+    bidObject.floor_price = 0;
   }
 
```

Comment:
I think it would be better to add the following params only if the mediaType is video. There's no reason to even try accessing the params for banner ads.

Please, add the params that are used for a specific mediaType conditionally.